src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -41,7 +41,8 @@
   <Target Name="_ComputeAssembliesToCompileToNative" DependsOnTargets="$(IlcDynamicBuildPropertyDependencies)">
 
     <Warning Condition="Exists($(UserRuntimeConfig))" Text="The published project has a runtimeconfig.template.json that is not supported by PublishAot. Move the configuration to the project file using RuntimeHostConfigurationOption." />
-    
+    <Warning Condition="'@(IlcRootedAssembliesExports)' != '' and '$(NativeLib)' == '' and '$(CustomNativeMain)' != 'true'" Text="Exporting methods from referenced assemblies is only available in the library or custom native main mode." />
```

Comment:
Why do we need to limit this like this? This option should be valid in all configurations.

Reply:
The limitation was set based on my understanding of the reported issue, but I could be wrong.

Could you please confirm (for my better understanding) whether the following scenario can be considered as a possible use case - when the application is compiled as an executable:
1. There is a managed framework - exposing unmanaged entry points
2. There is a managed app - exposing unmanaged entry points, referencing 1), and pinvoking into 3)
3. There is a native library - reverse pinvoking into both 1) and 2)

Reply:
Yes, the scenario is that the native library calls named unmanaged entrypoints defined in a managed library.

The type of the project does not matter: It is equally applicable to all app, app with custom main or dynamic library.

Reply:
Understood, thank you for the explanation.